boto3/resources/factory.py

```diff
@@ -133,9 +133,10 @@ def load_from_definition(self, resource_name,
 
         base_classes = [ServiceResource]
         if self._emitter is not None:
-            self._emitter.emit('creating-resource-class.%s' % cls_name,
-                               class_attributes=attrs,
-                               base_classes=base_classes)
+            self._emitter.emit(
+                'creating-resource-class.%s' % cls_name,
+                class_attributes=attrs, base_classes=base_classes,
+                service_context=service_context, emitter=self._emitter)
```

Comment:
I think I am fine with this change. Having the ``service_context`` is a quite informative bit of information that I could see used in helping create the action or base class and the ``emitter`` would seem useful to add extensibility logic when creating a method/class such that you could emit and register your own events on the creation of class/method.

@jamesls what are your thoughts on this?